### ISSUE TITLE
Continuous classifiers stuck in "final dispersion estimates"

### ISSUE THREAD

**Issue by ahdee** [dncR/MLSeq#6]
I have a pretty big dataset, about 3.4K genes and 1496 samples.  I find that the discrete and even the voom-based classifiers work well however any of the continous classifiers like random forest or svm gets stuck at the final dispersion estimates. The first two steps gene-wise dispersion estimates and mean-dispersion relationship gets computed fairly quickly but then it just hangs on the final dispersion estimates. I even let it run overnight but nothing seems to be happening? Any clues? 

here is the code for random forest 

```
rf <- classify(data = data.trainS4, method = "rf",
               preProcessing = "deseq-vst", ref = test_group ,
               control = trainControl(method = "repeatedcv", number = 5,
                                      repeats = 10, classProbs = TRUE))

```

for svm 

```
ctrl.continuous <- trainControl(method = "repeatedcv", number = 5, repeats = 10)
fit.svm <- classify(data = data.trainS4, method = "svmRadial",
                    preProcessing = "deseq-vst", ref = test_group, tuneLength = 10,
                    control = ctrl.continuous)

```
thanks.

**Comment by dncR** [comment]
Hello ahdee,

The data preprocessing steps (e.g., normalization, transformation, variance filtering, etc.) are common for each classifier even though some of them do not require variance stabilizing transformation such as Poisson Linear Discriminant analysis. If you stuck at the final dispersion estimates, this might be due to the functions and computations imported from other Bioconductor packages, e.g., DESeq2, limma, etc. The sample size and the number of features are large; however, it should have been completed overnight. We do not touch or modify the functions in dispersion estimate step. I believe you should externally run the dispersion estimate functions available in DESeq2 and limma packages. If it is able to complete dispersion estimates in less time, we might then go through the code line in MLSeq package to find its origin.

Best,
Dincer.

**Comment by ahdee** [comment]
@dncR  thanks for your help. I tried tracing it back step by step and follow to its internal function, what I found was that it was not the dispersion step rather than the classifier was just taking a really long time.  May be had it printed something like "normalization completed, start train"... or something like that it would had been easier to catch?  With that said, is there a function to skip normalization once its completed once, that is, the ability to feed just feed in the normalized object once a single continous classifier had been run? That would save a small bit of computation.
